src/main/Main.hs

```diff
@@ -283,16 +283,9 @@ commandLineHandler progName isInterpreter = complicatedOptions
                     updateCmd
                     (pure ())
         addCommand' "upgrade"
-                    "Upgrade to the latest stack (experimental)"
+                    "Upgrade to the latest stack"
                     upgradeCmd
-                    ((,) <$> switch
-                              ( long "git"
-                             <> help "Clone from Git instead of downloading from Hackage (more dangerous)" )
-                         <*> strOption
-                              ( long "git-repo"
-                             <> help "Clone from specified git repository"
-                             <> value "https://github.com/commercialhaskell/stack"
-                             <> showDefault ))
+                    (upgradeOpts Nothing) -- FIXME insert the default platform to use
```

Comment:
We need to figure out the right way to pass in the platform string (e.g., `osx-x86_64`). I see two options:

1. Pass in some CPP variable during official builds, and have only official builds use binary upgrade by default (probably the right approach).
2. Use the build system info from the Cabal package or similar to construct the string. Overall I'm -1 on this due to likely breaking cross-compilation and possibly having an incorrect configuration.